src/main/java/com/autosale/shop/service/impl/JwtTokenServiceImpl.java

```diff
@@ -0,0 +1,103 @@
+package com.autosale.shop.service.impl;
+
+import com.autosale.shop.model.JwtTokensDAO;
+import com.autosale.shop.model.User;
+import com.autosale.shop.model.UserRole;
+import com.autosale.shop.service.JwtTokenService;
+import com.autosale.shop.service.UserService;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.UnsupportedJwtException;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class JwtTokenServiceImpl implements JwtTokenService {
+
+    private final String SECRET_KEY;
+
+    private final UserService userService;
+
+    private final int ACCESS_TOKEN_LIFETIME;
+
+    private final int REFRESH_TOKEN_LIFETIME;
+
+    public JwtTokenServiceImpl(@Value("${jwt.secret}") String secret_key, UserService userService, @Value("${jwt.access_token_lifetime}") int access_token_lifetime, @Value("${jwt.refresh_token_lifetime}") int refresh_token_lifetime) {
+        SECRET_KEY = secret_key;
+        this.userService = userService;
+        ACCESS_TOKEN_LIFETIME = access_token_lifetime;
+        REFRESH_TOKEN_LIFETIME = refresh_token_lifetime;
+    }
+
+    @Override
+    public JwtTokensDAO generateTokensPair(User user) {
+        if (user.getId() == null) {
+            user = userService.getVerifiedUser(user.getUserName(), user.getPassword());
+        }
+        JwtBuilder jwtBuilder = Jwts.builder().claim("id", user.getId().toString())
+                .claim("username", user.getUserName())
+                .claim("role", user.getRole())
+                .setId(UUID.randomUUID().toString())
+                .setSubject(String.format("%s,%s", user.getId(), user.getUserName()))
+                .setIssuedAt(new Date()).setExpiration(Date.from(Instant.now().plus(ACCESS_TOKEN_LIFETIME, ChronoUnit.MINUTES)))
+                .signWith(Keys.hmacShaKeyFor(SECRET_KEY.getBytes()));
+        String accessToken = jwtBuilder.compact();
```

Comment:
why do you need this variable?

Reply:
Because then I change the builder settings to generate a refresh token, and transferring all changes directly to the constructor will, in my opinion, reduce the readability of the code